backend/src/packages/chats/chat.service.ts

```diff
@@ -83,8 +83,24 @@ class ChatService implements Service {
     return Promise.resolve(null);
   }
 
-  public delete(): ReturnType<Service['delete']> {
-    return Promise.resolve(true);
+  public async delete({
+    id,
+    userId,
+  }: {
+    id: number;
+    userId: number;
+  }): Promise<boolean> {
+    const allChats = await this.findAllByUserId(userId);
+
```

Comment:
maybe it is better to do smth like this:
    const chat = await this.find(id); 
    if (chat.userId !== userId) {
      throw new ChatError({
        status: HTTPCode.BAD_REQUEST,
        message: ExceptionMessage.INCORRECT_CREDENTIALS,
      });
    }
    return Boolean(await this.chatRepository.delete(id));